Cython/Utility/StringTools.c

```diff
@@ -176,6 +176,15 @@ static CYTHON_INLINE int __Pyx_UnicodeContainsUCS4(PyObject* unicode, Py_UCS4 ch
 
     }
 #endif
+    /* This should only happen if the user has manually created a non-PEP393 string
+     * (i.e. it's very unlikely)
```

Comment:
wchar strings are actually a common thing on Windows, where that's kind-of the native system Unicode string type. Windows file names fall into that category, for example.